frontend/controller/actions/identity.js

```diff
@@ -770,15 +771,23 @@ export default (sbp('sbp/selectors/register', {
           : { billableContractID: identityContractID }
         return [cid, credential]
       }))
-      await sbp('chelonia/fileDelete', manifestCids, credentials)
+      deleteResult = await sbp('chelonia/fileDelete', manifestCids, credentials)
     }
 
     if (shouldDeleteToken) {
       await sbp('gi.actions/identity/removeFileDeleteToken', {
         contractID: identityContractID,
-        data: { manifestCids }
+        data: {
+          manifestCids: manifestCids.filter((_, i) => {
+            return deleteResult[i].status === 'fulfilled'
+          })
+        }
       })
     }
+
+    if (deleteResult.some(r => r.status === 'rejected')) {
+      throw new Error('[gi.actions/identity/removeFiles] Some CIDs could not be deleted')
+    }
```

Comment:
Note that throwing in a sideEffect will show the user a very serious unrecoverable red error global banner ... are you sure you want to throw? Maybe log the error instead?